src/main/java/org/openrewrite/java/spring/boot2/WebSecurityConfigurerAdapterRecipe.java

```diff
@@ -0,0 +1,240 @@
+/*
+ * Copyright 2022 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java.spring.boot2;
+
+import org.openrewrite.*;
+import org.openrewrite.internal.ListUtils;
+import org.openrewrite.java.*;
+import org.openrewrite.java.search.UsesType;
+import org.openrewrite.java.tree.*;
+import org.openrewrite.marker.Markers;
+
+import java.util.*;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.stream.Collectors;
+
+/**
+ * @author Alex Boyko
+ */
+public class WebSecurityConfigurerAdapterRecipe extends Recipe {
+
+    private static final Collection<J.Modifier.Type> EXPLICIT_ACCESS_LEVELS = Arrays.asList(J.Modifier.Type.Public,
+            J.Modifier.Type.Private, J.Modifier.Type.Protected);
+
+    private static final String FQN_CONFIGURATION = "org.springframework.context.annotation.Configuration";
+    private static final String FQN_WEB_SECURITY_CONFIGURER_ADAPTER = "org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter";
+    private static final String FQN_SECURITY_FILTER_CHAIN = "org.springframework.security.web.SecurityFilterChain";
+    private static final String FQN_OVERRIDE = "java.lang.Override";
+    private static final String FQN_WEB_SECURITY_CUSTOMIZER = "org.springframework.security.config.annotation.web.configuration.WebSecurityCustomizer";
+    private static final String BEAN_PKG = "org.springframework.context.annotation";
+    private static final String BEAN_SIMPLE_NAME = "Bean";
+    private static final String FQN_BEAN = BEAN_PKG + "." + BEAN_SIMPLE_NAME;
+    private static final String BEAN_ANNOTATION = "@" + BEAN_SIMPLE_NAME;
+
+    private static final MethodMatcher WEB_SECURITY_CONFIGURER_ADAPTER_METHOD_MATCHER =
+            new MethodMatcher(FQN_WEB_SECURITY_CONFIGURER_ADAPTER + " *(..)", true);
```

Comment:
this is unnecessary if `TypeUtils#isAssignableTo` is used for checking the methods declaring type.